src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -301,7 +301,10 @@ private ParseUserDefinitionResult ParseUDFsAndNamedFormulas(string script, Parse
                     {
                         CreateError(_curs.TokCur, TexlStrings.ErrUDF_MissingReturnType);
                         udfs.Add(new UDF(thisIdentifier.As<IdentToken>(), colonToken, returnType: null, new HashSet<UDFArg>(args), body: null, _hasSemicolon, parserOptions.NumberIsFloat, isValid: false));
-                        break;
+
+                        // Return type not found, move to next user definition
+                        MoveToNextUserDefinition();
+                        continue;

```

Comment:
Why do we need this for return type?

Reply:
When a return type is not found it's still considered a parser error and we must move to the next user definition.